### PR TITLE
Working on stabilization

### LINKED ISSUE #144: Create a `target_straight` function
Reaches a target position/speed in a straight line accelerating as much as possible and decelerating just in time (i.e.: as fast as possible).

### LINKED ISSUE #181: Cul-de-sac corrections
This is important as turn errors might accumulate. In example: turn back at goal (2 right), turn right (1), turn back (2 right), turn right (1)... That is a total of 6 consecutive turns ending in a cell with no lateral walls:

```
o---o---o---o---o
|           | G |
o   o   o   o   o
|   |   |       |
o   o   o---o   o
|   |           |
o   o---o   o---o
|   |           |
o   o   o   o   o
|       |   |   |
o   o   o   o   o
| S |   |       |
o---o---o---o---o
```

Gyroscope would probably help a lot with this.

http://www.micromouseonline.com/2011/09/13/correct-for-errors-in-a-cul-de-sac/

Execute them only when there is an actual difference between side/front sensor readings before turning back.

- First front correction (make sure we are perpendicular and at a known distance).
- Turn 90º in the smarter direction:
  - No lateral walls: random.
  - Single lateral wall: to that lateral wall.
  - Both lateral walls: the one that is further away.
- Another front correction (we should be perpendicular, but check, and correct distance).
- Turn 90º.
- Optionally, move back a bit.

### LINKED ISSUE #259: Review calibration
- Calibration functions:
  - Front sensors calibration should end at the expected position.
  - Front sensors calibration and micrometers-per-meter calibration should not block the robot after the test.
- Movements:
  - Stop as expected at the middle, beginning or ending of a cell.
  - End turns at the expected point as well.

### PR DESCRIPTION
**Pull request by Peque**
Better acceleration functions. 

Fixes #144. Fixes #181. Fixes #259.

### CODE REVIEW

File: src/target/move.c [hunk 1]
```diff
@@ -119,19 +126,48 @@ void disable_walls_control(void)
...
+			   (int32_t)required_micrometers_to_speed(speed))
+			;
+	} else {
+		set_target_linear_speed(-max_linear_speed);
+		while (get_encoder_average_micrometers() >
+		       target_distance +
+			   (int32_t)required_micrometers_to_speed(speed))
+			;
+	}
+	change_speed(speed);
```
Comment (cua-cua) [inline src/target/move.c]: It would be clear to do it without change_speed function

File: src/target/move.c [hunk 2]
```diff
@@ -250,33 +250,38 @@ void move_front(void)
 }
 
 /**
- * @brief Move left into the next cell.
+ * @brief Move left or right into the next cell.
  */
-void move_left(void)
+static void move_side(enum step_direction side)
```
Comment (cua-cua) [inline src/target/move.c]: Create an issue to configure these variables.

File: src/target/move.c [hunk 3]
```diff
@@ -238,6 +276,44 @@ void turn_right(void)
...
+{
+	if (read_cycle_counter() % 2)
+		return RIGHT;
+	return LEFT;
+}
+
+/**
+ * @brief Turn back (180-degree turn) and correct with front walls if possible.
+ */
+void turn_back_with_corrections(void)
```
Comment (cua-cua) [inline src/target/move.c]: I would delete "with corrections"

File: src/target/move.c [hunk 4]
```diff
@@ -304,11 +304,19 @@ void turn_back_with_corrections(void)
 {
 	enum step_direction side;
 
-	keep_front_wall_distance(CELL_DIMENSION / 2., 0.001);
 	side = choose_random_side();
```
Comment (cua-cua) [inline src/target/move.c]: delete it

File: src/target/move.c (hunk 1 above)
Comment (Peque) [reply src/target/move.c]: Removed. :+1:

File: src/target/move.c (hunk 3 above)
Comment (Peque) [reply src/target/move.c]: 2a7025cb17ac3f8d1012aee421c54259cbdef0c6

File: src/target/move.c (hunk 1 above)
Comment (Peque) [reply src/target/move.c]: 5a974fda1f47e6f16b6bfb0c13487e17f4060624

File: src/target/move.c (hunk 2 above)
Comment (Peque) [reply src/target/move.c]: https://github.com/Theseus/bulebule/issues/286

File: src/target/move.c (hunk 4 above)
Comment (Peque) [reply src/target/move.c]: f915452351423a7f9862bc61e4007b0ec62f7dc8

**Pull request merged**